Firebase/InstanceID/FIRInstanceID.m

```diff
@@ -893,10 +896,19 @@ - (void)fetchDefaultToken {
   [self defaultTokenWithHandler:handler];
 }
 
-- (void)defaultTokenWithHandler:(FIRInstanceIDTokenHandler)handler {
+- (void)defaultTokenWithHandler:(nullable FIRInstanceIDTokenHandler)aHandler {
+  // Store handlers to be called once operation completes
+  if (!self.defaultTokenFetchHandler) {
+    self.defaultTokenFetchHandler = [[FIRInstanceIDCombinedHandler<NSString *> alloc] init];
+  }
+  if (aHandler) {
+    [self.defaultTokenFetchHandler addHandler:aHandler];
+  }
+
   if (self.isFetchingDefaultToken || self.isDefaultTokenFetchScheduled) {
```

Comment:
I feel like you can also get rid of isDefaultTokenFetchScheduled. Because you have the first original call back in and retry just calling with nil.

But if you don't feel comfortable, you can leave it like this for now.

Reply:
As far as I can see, currently `isDefaultTokenFetchScheduled` prevents additional requests to be sent while `[self retryIntervalToFetchDefaultToken]` timeout, so, I think, we still need it.

Reply:
I think since we are adding a new logic to the existing code that could replace the old boolean value. We should do that. We are trying to simplify the iid logic especially your new queue should be able to handle it.

So one way we can do is to have a defaultTokenWithRetry:(BOOL) handler:
So that if it's retry, we always go through, without adding a new handler to the queue. Otherwise, adding it to the queue and check if it's >1.

Reply:
I pushed an update. Would you take a look if it is what you meant?